### PR TITLE
fix: call EqualVT in Equal

### PR DESCRIPTION
**Pull request by janisz**
### Description

This PR introduces Equalable interface and use it in `protocompat.Equal` so instead of relaying on `proto.Equal` implementation we can use generated `EqualVT` method that does not rely on reflection.

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed
### Testing
- [x] inspected CI results
#### Automated testing
- [x] modified existing tests
- [x] contributed **no automated tests**
#### How I validated my change
CI

### CODE REVIEW

File: pkg/protocompat/time.go [hunk 1]
```diff
@@ -211,5 +211,9 @@ var (
 
 // IsZeroTimestamp returns whether a Timestamp pointer is either nil, or pointing to the zero of the type.
 func IsZeroTimestamp(ts *timestamppb.Timestamp) bool {
-	return ts == nil || Equal(ts, GetProtoTimestampZero()) || Equal(ts, zeroProtoTimestampFromTime)
+	return ts == nil || equal(ts, GetProtoTimestampZero()) || equal(ts, zeroProtoTimestampFromTime)
```
Comment (mtodor) [inline pkg/protocompat/time.go:214]: Can we use `CompareTimestamps` instead of `equal`?

File: pkg/protocompat/time.go [hunk 2]
```diff
@@ -211,5 +211,9 @@ var (
...
+}
+
+func equal(a, b *timestamppb.Timestamp) bool {
+	return a.Nanos == b.Nanos && a.Seconds == b.Seconds
```
Comment (rhybrillou) [inline pkg/protocompat/time.go:218]:
```suggestion
	return a.AsTime().Equal(b.AsTime())
```

**Review by rhybrillou — APPROVED** [review]

File: pkg/protomock/mock_matcher.go [hunk 3]
```diff
@@ -5,8 +5,12 @@ import (
 	"go.uber.org/mock/gomock"
 )
 
-func GoMockMatcherEqualMessage(expectedMsg protocompat.Message) gomock.Matcher {
+func GoMockMatcherEqualMessage[T protocompat.Equalable[T]](expectedMsg T) gomock.Matcher {
 	return gomock.Cond(func(msg any) bool {
-		return protocompat.Equal(expectedMsg, msg.(protocompat.Message))
+		return protocompat.Equal(expectedMsg, toT(expectedMsg, msg))
```
Comment (mtodor) [inline pkg/protomock/mock_matcher.go:10]: Can we cast to `T` directly without a helper function?
```suggestion
		return gomock.Cond(func(msg any) bool {
		return protocompat.Equal(expectedMsg, msg.(T))
	})
```

**Review by mtodor — APPROVED** [review]

**Pull request merged**